test/filters/test_functor_filter.cpp

```diff
@@ -94,6 +94,59 @@ INSTANTIATE_TEST_SUITE_P(RandomSeed,
                          FunctorFilterRandom,
                          testing::Values(123, 456, 789));
 
+
+template <typename T>
```

Comment:
This test is almost a copy paste. We need to think how we should test executor using code

Reply:
Indeed, might as well try to extend the original test and reuse as much code as possible. 

besides being a carbon copy of the original test, you're testing with a single seed and you need need to always validate behavior with 2 or more seeds.

Reply:
Combined the test 99c86ee.
Not the most elegant way but get the job done. It tests using multiple seeds all the executors as well as check best fit.